### PR TITLE
Auto DJ short track

### PR DESCRIPTION
**Pull request by daschuer**
This fixes: 
https://bugs.launchpad.net/mixxx/+bug/1396385

Now, tracks shorter than transition time do not stop AutoDJ anymore.

### CODE REVIEW

File: src/library/autodj/autodjprocessor.cpp [hunk 1]
```diff
@@ -485,13 +495,17 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
...
                 // fadeDuration next).
-                calculateFadeThresholds(&otherDeck);
+                calculateTransition(&otherDeck, &thisDeck);
 
                 // For negative fade durations, jump back to insert a pause
                 // between the tracks.
                 if (thisFadeDuration < 0.0) {
-                    // TODO(rryan) why otherDeck.fadeDuration?
+                    // Note: since the fade duration is relative to the track
+                    // length, we need to user here the other deck fade duration
```
Comment (rryan) [inline src/library/autodj/autodjprocessor.cpp]: s/user here/use/

**Comment by rryan** [comment]
Please add a test or two that shows this is fixed.

**Comment by rryan** [comment]
Please add a test directly to the 1.12 branch -- AutoDJ deserves heavy testing and we already have a comprehensive test suite so this reduces test coverage.

**Comment by daschuer** [comment]
Thank you for Review. The Test is already on my TODO list.

**Pull request merged**